pkg/config/config_template.yaml

```diff
@@ -21,6 +21,19 @@ api_key:
 #
 # dd_url: https://app.datadoghq.com
 
+## @param cloud_provider - string - optional - default: <empty>
+## This option restrict which endpoint will be used by the agent to retrieve host metadata. By default the agent will try
+## all cloud provider.
+## Possible values are:
+## "AWS"     AWS EC2, ECS/Fargate
+## "GCP"     Google Cloud Provider
+## "Azure"   Azure
+## "Alibaba" Alibaba
```

Comment:
I think we should allow the case of these values to differ (for example interpret `"gcp"` or `"alibaba"` as valid values), to prevent the case from confusing users.